File: pinyougou_manager_web/src/main/webapp/js/service/specificationService.js
app.service("specificationService",function ($http) {

    this.findPage=function (pageNumber,pageSize,searchEntity) {
        return $http.post("../specification/findPage.do?pageNumber="+pageNumber+"&pageSize="+pageSize,searchEntity);
    };
    this.add=function (entity) {
        return $http.post("../specification/add.do",entity);
    };
    this.update=function (entity) {
        return $http.post("../specification/update.do",entity);
    };
    this.findOne=function (id) {
        return $http.get("../specification/findOne.do?id="+id);
    };
    this.dele=function (ids) {
        return $http.get("../specification/dele.do?ids="+ids);
    };
    this.selectSpecOptions=function () {
        return $http.get("../specification/selectSpecOptions.do?");
    }
});

File: pinyougou_sellergoods_interface/src/main/java/com/pinyougou/sellergoods/service/SpecificationService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbSpecification;
import com.pinyougou.pojo.TbSpecificationExample;
import entity.PageResult;
import groupEntity.Specification;

import java.util.List;
import java.util.Map;

public interface SpecificationService {
    PageResult search( TbSpecification tbSpecification, int page, int size);

    List<TbSpecification> findAll(TbSpecificationExample example);

    TbSpecification findById(Long id);

    void save(Specification specification);

    void update(Specification specification);

    void delete(Long id);

    PageResult findPage(int pageNumber, int pageSize);

    Specification findOne(Long id);

    List<Map> selectSpecOptions();

}
